internal/oc/release_test.go

```diff
@@ -570,45 +572,222 @@ var _ = Describe("getIcspFileFromRegistriesConfig", func() {
-		icspFile, err := oc.getIcspFileFromRegistriesConfig(log)
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(icspFile).ShouldNot(Equal(""))
-		data, err := os.ReadFile(icspFile)
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(string(data)).Should(Equal(expected))
-	})
+	Context("With IDSM support", func() {
+		BeforeEach(func() {
+			mockExecuter.EXPECT().Execute("oc", []string{"adm", "release", "info", "--help"}).Return(
+				`--idms-file='':`, "", 0,
+			).AnyTimes()
+		})
+
+		It("One valid mirror registry", func() {
+			regData := []mirrorregistries.RegistriesConf{
+				{
+					Location: "registry.ci.org",
+					Mirror: []string{
+						"host1.example.org:5000/localimages",
+					},
+				},
+				{
+					Location: "quay.io",
+					Mirror: []string{
+						"host1.example.org:5000/localimages",
+					},
+				},
+			}
+			mockMirrorRegistriesConfigBuilder.EXPECT().IsMirrorRegistriesConfigured().Return(true).Times(1)
+			mockMirrorRegistriesConfigBuilder.EXPECT().ExtractLocationMirrorDataFromRegistries().Return(regData, nil).Times(1)
+			mirrorsFlag, err := oc.getMirrorsFlagFromRegistriesConfig(log)
+			Expect(err).ShouldNot(HaveOccurred())
+			defer mirrorsFlag.Delete()
+			Expect(mirrorsFlag).ShouldNot(BeNil())
+			Expect(mirrorsFlag.file).ShouldNot(BeEmpty())
+			data, err := os.ReadFile(mirrorsFlag.file)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(data).To(MatchYAML(dedent.Dedent(`
+				apiVersion: config.openshift.io/v1
+				kind: ImageDigestMirrorSet
+				metadata:
+				  creationTimestamp: null
+				  name: image-policy
+				spec:
+				  imageDigestMirrors:
+				  - mirrors:
+				    - host1.example.org:5000/localimages
+				    source: registry.ci.org
+				  - mirrors:
+				    - host1.example.org:5000/localimages
+				    source: quay.io
+				status: {}
+			`)))
+		})
+
+		It("Multiple valid mirror registries", func() {
+			regData := []mirrorregistries.RegistriesConf{
+				{
+					Location: "registry.ci.org",
+					Mirror: []string{
+						"host1.example.org:5000/localimages",
+						"host1.example.org:5000/openshift",
+					},
+				},
+				{
+					Location: "quay.io",
+					Mirror: []string{
+						"host1.example.org:5000/localimages",
+					},
+				},
+			}
+			mockMirrorRegistriesConfigBuilder.EXPECT().IsMirrorRegistriesConfigured().Return(true).Times(1)
+			mockMirrorRegistriesConfigBuilder.EXPECT().ExtractLocationMirrorDataFromRegistries().Return(regData, nil).Times(1)
+			mirrorsFlag, err := oc.getMirrorsFlagFromRegistriesConfig(log)
+			Expect(err).ShouldNot(HaveOccurred())
+			defer mirrorsFlag.Delete()
+			Expect(mirrorsFlag).ShouldNot(BeNil())
+			Expect(mirrorsFlag.file).ShouldNot(BeEmpty())
+			data, err := os.ReadFile(mirrorsFlag.file)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(data).Should(MatchYAML(dedent.Dedent(`
+				apiVersion: config.openshift.io/v1
+				kind: ImageDigestMirrorSet
+				metadata:
+				  creationTimestamp: null
+				  name: image-policy
```

Comment:
Am I being picky to suggest that even in the test, we should not call this a policy?

Same comment for all other naming within ImageDigestMirrorSet tests

Reply:
Will rename it to "image-mirror-set". Does that look better? Any other suggestion? Note that these (both ICSP and IDMS) are written to temporary files, only for use while running the various "oc" commands, after that they are deleted. So the name isn't really important (unless I miss something).